### PR TITLE
Review possible explanatory realtime situations

### PR DESCRIPTION
**Pull request by r3to**
- [x] Compare implementation to Realisierungsvorgabe Profil CH SIRI-SX/VDV736 and Android
- [x] Add TripStatus (wip)
- [x] Test Situations (see https://github.com/openTdataCH/ojp-sdk/issues/41)
- [x] Update Sample App

### CODE REVIEW

**Comment by r3to** [comment]
To be discussed: do we want to augment the `timedLegs` with the corresponding `PTSituations` in order to avoid to have to do the mapping in the client?

File: SampleApp/OJPSampleApp/Helpers/OJP+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        case .vehicleFailure:
+            "Fahrzeugstörung"
+        case .poorWeather:
+            "Unwetter"
+        case .routeBlockage:
+            "Blockierte Strecke"
+        case .technicalProblem:
+            "Tchnisches Problem"
+        case .unknown:
+            "Störung"
```
Comment (terencebls) [inline SampleApp/OJPSampleApp/Helpers/OJP+Extensions.swift:128]: Maybe in SDK we can write "Unknown" so we don't have the impression that it works.

File: Sources/OJP/ConvenienceExtensions/TripRequest+Extensions.swift
Comment (terencebls) [inline Sources/OJP/ConvenienceExtensions/TripRequest+Extensions.swift:1]: Maybe we can move those stuff in OJP+Extensions?

Comment (r3to) [reply Sources/OJP/ConvenienceExtensions/TripRequest+Extensions.swift:1]: Those are public extensions in the SDK that can be used by the client, while OJP+Extensions are only in the SampleApp's target. 
Maybe there are some in of those, that would be convenient to move to the SDK.

File: SampleApp/OJPSampleApp/Views/TripDetailView.swift
Comment (terencebls) [inline SampleApp/OJPSampleApp/Views/TripDetailView.swift:1]: image

Comment (terencebls) [inline SampleApp/OJPSampleApp/Views/TripDetailView.swift:1]: image
 Same Situations twice (but maybe are the data false?)

Comment (r3to) [reply SampleApp/OJPSampleApp/Views/TripDetailView.swift:1]: ja, müsste schon `timetabledTime` sein.

In der Übersichts Liste zeigen wir die `StartTime` welche ebenfalls `estimatedTime` ist und nicht timetabled.. 
image

Comment (r3to) [reply SampleApp/OJPSampleApp/Views/TripDetailView.swift:1]: thanks, yes, there is an issue, where allPTsituations is not checked for duplicates

**Review by terencebls — APPROVED** [review]

**Pull request merged**